rotkehlchen/exchanges/bitmex.py

```diff
@@ -212,67 +235,49 @@ def _api_query(
             )
 
         try:
-            json_ret = json.loads(response.text)
+            json_ret: list = json.loads(response.text)
         except JSONDecodeError as e:
             raise RemoteError('Bitmex returned invalid JSON response') from e
 
-        if isinstance(json_ret, dict) and 'error' in json_ret:
-            raise RemoteError(json_ret['error']['message'])
-
         return json_ret
 
-    def _api_query_dict(
-            self,
-            verb: str,
-            path: str,
-            options: dict | None = None,
-    ) -> dict:
-        result = self._api_query(verb, path, options)
-        assert isinstance(result, dict)  # pylint: disable=isinstance-second-argument-not-valid-type
-        return result
-
-    def _api_query_list(
-            self,
-            verb: str,
-            path: str,
-            options: dict | None = None,
-    ) -> list:
-        result = self._api_query(verb, path, options)
-        assert isinstance(result, list)  # pylint: disable=isinstance-second-argument-not-valid-type
-        return result
-
     @protect_with_lock()
     @cache_response_timewise()
-    def query_balances(self) -> ExchangeQueryBalances:
+    def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+        self.first_connection()
         returned_balances: dict[AssetWithOracles, Balance] = {}
         try:
-            resp = self._api_query_dict('get', 'user/wallet', {'currency': 'XBt'})
-            # Bitmex shows only BTC balance
-            usd_price = Inquirer.find_usd_price(self.btc)
+            resp = self._api_query('user/wallet', {'currency': 'all'})
         except RemoteError as e:
             msg = f'Bitmex API request failed due to: {e!s}'
             log.error(msg)
             return None, msg
 
-        # result is in satoshis
-        try:
-            amount = satoshis_to_btc(deserialize_asset_amount(resp['amount']))
-        except DeserializationError as e:
-            msg = f'Bitmex API request failed. Failed to deserialized amount due to {e!s}'
-            log.error(msg)
-            return None, msg
+        for balance in resp:
+            try:
+                asset = bitmex_to_world(currency := balance['currency'])
+                raw_amount = balance['amount']
+            except (KeyError, UnknownAsset) as e:
+                log.error(e)
+                continue
 
-        usd_value = amount * usd_price
-        returned_balances[self.btc] = Balance(
-            amount=amount,
-            usd_value=usd_value,
-        )
-        log.debug(
-            'Bitmex balance query result',
-            currency='BTC',
-            amount=amount,
-            usd_value=usd_value,
-        )
+            if (decimals := self.asset_to_decimals.get(currency)) is None:
+                log.error(f'Unknown decimals for asset balance {balance} in bitmex. Skipping')
```

Comment:
Add asset name / symbol in this message

Reply:
it is included in balance. Balance is a mapping that contains all the info